scrapli_netconf/driver/base_driver.py

```diff
@@ -768,26 +772,88 @@ def _pre_delete_config(self, target: str = "running") -> NetconfResponse:
         )
         return response
 
-    def _pre_commit(self) -> NetconfResponse:
+    def _pre_commit(
+        self,
+        confirmed: bool = False,
+        timeout: Optional[int] = None,
+        persist: Optional[int] = None,
+        persist_id: Optional[int] = None,
+    ) -> NetconfResponse:
         """
         Handle pre "commit" tasks for consistency between sync/async versions
 
         Args:
-            N/A
+            confirmed: whether this is a confirmed commit
+            timeout: specifies the confirm timeout in seconds
+            persist: make the confirmed commit survive a session termination, and set a token on
+                the ongoing confirmed commit
+            persist_id: value must be equal to the value given in the <persist> parameter to the
+                original <commit> operation.
 
         Returns:
             NetconfResponse: scrapli_netconf NetconfResponse object containing all the necessary
                 channel inputs (string and xml)
 
         Raises:
-            N/A
+            ScrapliValueError: if persist and persist_id are provided (cannot combine)
+            ScrapliValueError: if confirmed and persist_id are provided (cannot combine)
+            CapabilityNotSupported: if device does not have confirmed-commit capability
 
         """
         self.logger.debug("Building payload for 'commit' operation")
         xml_request = self._build_base_elem()
         xml_commit_element = etree.fromstring(
             NetconfBaseOperations.COMMIT.value, parser=self.xml_parser
         )
+
+        if persist and persist_id:
+            raise ScrapliValueError(
+                "Invalid combination - 'persist' cannot be present with 'persist-id'"
+            )
+        if confirmed and persist_id:
+            raise ScrapliValueError(
+                "Invalid combination - 'confirmed' cannot be present with 'persist-id'"
+            )
+
+        if confirmed or persist_id:
```

Comment:
all my changes were cosmetic except this one (well I hope this one is too) -- does this still look correct/accurate to you. I just moved the capability checking from within each of the `confirmed` and `persist_id` blocks up here so we can do it in only one place.